Automattic/jetpack
# Charts: Refactor Tooltips

## Proposed changes:


This PR introduces a refactor of the tooltip implementation in the charts library. It focuses on improving component modularity, styling, accessibility, and extensibility. These changes address previous PR feedback, known limitations and align with best practices for performance and maintainability.

  - Introduced `BaseTooltip` as a reusable, foundational tooltip component.
  - Enabled custom tooltip content via a `component` prop.
  - Added accessibility attributes for better usability.
  - Replaced `interface` definitions with `type` for better extensibility.
  - Moved types into a dedicated `types.ts` file for organization.
  - Refactored event handling for cleaner and more efficient logic.
  - Removed inline styles from components (kept in stories for illustrative purposes).
  - Introduced SCSS modules for scoped styling.
  - Updated tooltips for the line chart and the pie semi circle chart
  - Restructured file exports to support multiple tooltip versions through a unified index file.

#### Future improvement:

Considering centralising tooltip state management and rendering logic across chart components into a reusable custom hook. This could include managing tooltip visibility, position, and data within the hook, as well as introducing a reusable tooltip render component to eliminate duplication and ensure consistency across the library. Mentioned in comment 
https://github.com/Automattic/jetpack/pull/40581/files#r1881899511


### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Install Jetpack packages if needed `pnpm i`
* navigate to the directory and run storybook: `cd projects/js-packages/charts` and `pnpm run storybook`
* verify that existing charts are working with tooltips all as expected
